r2dbc-mysql/src/main/java/SimpleStatement.kt

```diff
@@ -45,9 +47,40 @@ class SimpleStatement(private val clientSupplier: Supplier<JasyncConnection>, pr
         return this
     }
 
+    override fun returnGeneratedValues(vararg columns: String): Statement {
+        if (columns.any { !it.equals("LAST_INSERT_ID", true) }) {
+            throw IllegalArgumentException("MySQL only support return special generated value that is is 'LAST_INSERT_ID'")
+        }
+
+        if (sql.indexOf("INSERT", ignoreCase = true) <= 0) {
+            throw IllegalStateException("Statement must be a INSERT command when includes 'LAST_INSERT_ID'")
+        }
+
+        selectLastInsertId = true
+
+        return this
+    }
+
     override fun execute(): Publisher<out Result> {
         val jasyncConnection = clientSupplier.get()
+        val queried = queryExecute(jasyncConnection)
+
+        if (selectLastInsertId) {
+            return queried.flatMap { result ->
```

Comment:
Is it possible to combine `SELECT LAST_INSERT_ID()` with the insert statement? E.g. appending `SELECT LAST_INSERT_ID()` to the insert query so we get just a single server roundtrip?

Reply:
There are two checks for `selectLastInsertId`, a single check is sufficient. I also slightly changed the code:

```kotlin
         if (selectLastInsertId) {
            return queried.flatMap { result ->
                    jasyncConnection.sendQuery("SELECT LAST_INSERT_ID() AS $generatedKeyName")
                        .toMono()
                        .map { JaysncResult(it.rows, result.rowsAffected) }
            }
        }
```